core/services/network_checker.rb

```diff
@@ -66,16 +66,20 @@ def self.random_resources(resources)
 
   # Check single resource for availability by curl
   #  @return [Boolean] true if resource available
-  def self.check_resource_by_curl?(machine_configurator, machine, resource)
-    result = machine_configurator.run_command(machine, "curl -Is #{resource} | head -n 1 ")
-    curl_responce_successfull?(result.value)
+  def self.check_resource_by_curl?(machine_configurator, machine, resource, logger)
+    machine_configurator.within_ssh_session(machine) do |connection|
+      result = machine_configurator.ssh_exec(connection, "curl -Is #{resource} | head -n 1", logger)
+      return curl_responce_successfull?(result.value)
+    end
   end
 
   # Check single resource for availability by wget
   #  @return [Boolean] true if resource available
-  def self.check_resource_by_wget?(machine_configurator, machine, resource)
-    result = machine_configurator.run_command(machine, "wget -S -q --spider #{resource}")
-    !result.error?
+  def self.check_resource_by_wget?(machine_configurator, machine, resource, logger)
+    machine_configurator.within_ssh_session(machine) do |connection|
+      result = machine_configurator.ssh_exec(connection, "wget -S -q --spider #{resource}", logger)
+      return !result.error?
```

Comment:
Modify `MachineConfigurator#run_command` method, so it will have the optional logger argument. And then return to using it.